src/libmongoc/src/mongoc/mongoc-cluster-aws.c

```diff
@@ -1242,7 +1242,7 @@ _client_second (mongoc_cluster_t *cluster,
       AUTH_ERROR_AND_FAIL ("Failed to parse server nonce");
    }
 
-   if (!kms_request_append_payload (request, body, -1)) {
+   if (!kms_request_append_payload (request, body, strlen(body))) {
```

Comment:
@kevinAlbs: I can confirm that this avoids the assertion failure in PHPC. Authentication still fails for other reasons, which I'm investigating, but we do make it beyond this point.